Automattic/wp-calypso
# CRO i.2: integrate Jetpack Product Slide Out Card capability

### Changes proposed in this Pull Request

This PR makes it possible to have the Jetpack Product Slide Out Cards in the features section of the 3 Plans.

#### Notes:
This PR is a branch off of `add/product-card-i2-features`, [#46789](https://github.com/Automattic/wp-calypso/pull/46789) .
This working branch did not yet have the JetpackProductSlideOutCard component available yet, so in it's place is testing/placeholder markup displaying all the component values and props, whereas the `JetpackProductSlideOutCard` component and import statement is there, but it is commented out. Once this PR is merged, we can create another PR to remove the placeholder markup, and uncomment out the component markup, making it active.

This PR also does not handle how the product slide-out cards are displayed if/when products are already owned.
This PR is mainly just about making it possible to drop in the JetpackProductSlideOutCard component in the Features section and make it functional. There may be some slight design improvements that can be done in a following PR when we activate the JetpackProductSlideOutCard component.

#### Implementation Notes:

 - Created 2 constants (`FEATURE_TO_PRODUCT_ALT_V2`, `FEATURE_TO_MONTHLY_PRODUCT_ALT_V2`) to map the product features to the appropriate product slug.
 - Had to pass the product `billingTerm` and `onProductClick` prop through the `JetpackProductCardAlt2` component and into the `JetpackProductCardFeatures` component.
 - Created a special `constants.FEATURE_PRODUCT_SCAN_V2_NO_SLIDEOUT` feature key in order to be able to specifically exclude it from rendering the product slide-out card only in the Jetpack Security Real-time plan. (Instead of the slide-out card, it renders the info-popout).
- Modified the `JetpackProductCardFeaturesItem` component to render and toggle the "Learn more" & "Close" product dropdown toggle.
 - Created a new `FeaturesProductSlideOut` component containing the logic and rendering of the new `JetpackProductSlideOutCard` component.

### Testing instructions

The link to the Figma design mockup and prototype is located here in this Asana card:
1196341175636977-as-1198496807750941/f

1. Run this PR (both Calypso and Cloud)
2. Visit the `Plans" page in Calypso and the `/pricing` page in Jetpack Cloud.
3. Select a Jetpack site with Free Plan (No plans owned)
4. Expand all the Plans features sections.
5. Click to open each slide-out product and inspect the product info(props) and that it is correct for each product. Check both Monthly and Yearly.
6. Verify the "Learn more" and "Close" dropdown functionality works for all of them.
7. Verify that the "Scan" product for Jetpack Security Real-time does **_not_** have a product slide-out but instead an Info pop-out.

Screenshot:
![Markup 2020-10-27 at 20 39 20](https://user-images.githubusercontent.com/11078128/97388106-a1942880-1894-11eb-80db-049215295aca.png)


Fixes: 1196341175636977-as-1198930356649981/f
